### ISSUE TITLE
JitBuilder now requires VirtualMachineState

### ISSUE THREAD

**Issue by mstoodle** [eclipse/omr#623]
When I introduced support for modelling VM state variables in JitBuilder, it was not my intention to make them mandatory. But I achieved that anyway because the code that automatically propagates VM state along control flow edges trips an assertion if the source has a NULL vm state. Of course, if you don't allocate a vm state, it will be NULL.

There is a workaround: in your method builder's ::buildIL() method, before appending any byte code builder objects, do the following:
setVMState(new OMR::VirtualMachineState());

A better solution would be to gate the assertion only if the method builder has a non-NULL vmState (implying that we really want to model VM state so encountering a NULL VM state should be an error).

**Comment by charliegracie** [comment]
I agree that this should be optional.  I am not sure all users of JIT Builder will want / need to model VM state this way.

**Comment by Leonardo2718** [comment]
I think having a convenient way of modeling VM state is very useful to have. But it should definitely be optional as I can foresee some VMs having complex state that cannot be easily modeled using `VirtualMachineState`. It's a trade-off between simplicity and generality.

**Comment by mstoodle** [comment]
@Leonardo2718 I think the only required thing right now is the single line of code I put into the initial comment. Clients are required to model their entire VM state, even when they use the VM state services: you only need to model those parts that you want to "privatize" into the native stack frame and be able to reconstitute on transitions out of the native frame.

But I completely agree that VM state modelling should completely be opt-in. I'm hoping not to clutter the code a lot to manage that :) .

Thanks for the supportive feedback, guys!
